src/lib.rs

```diff
@@ -652,14 +652,61 @@ impl Config {
         self.statik.unwrap_or_else(|| self.infer_static(name))
     }
 
+    fn pkg_config_from_path(&self) -> OsString {
+        if cfg!(target_os = "windows") {
+            // Resolve pkg-config in PATH to find the absolute path to pkg-config that we should
+            // use, so that we can skip Strawberry Perl's pure-perl implementation of pkg-config,
+            // which (despite its presence in PATH) is for internal use and not meant to be used as
+            // a MinGW distribution.
+            let pathexts = env::var_os("PATHEXT").map_or_else(
+                || Vec::from(["COM", "EXE", "BAT"].map(|v| OsStr::new(v).to_owned())),
+                |v| {
+                    env::split_paths(&v)
+                        .map(|v| v.into_os_string())
+                        .collect::<Vec<_>>()
+                },
+            );
+            if let Some(paths) = env::var_os("PATH") {
+                for mut path in env::split_paths(&paths) {
+                    path.push("pkg-config");
+                    for pathext in &pathexts {
+                        path.set_extension(pathext);
+                        if !path.is_file() {
+                            continue;
+                        }
+                        // Strawberry Perl's pkg-config is pkg-config.bat
+                        if *pathext == "bat" || *pathext == "BAT" {
+                            let mut cmd = Command::new(path.as_os_str());
+                            let out = match cmd.output() {
+                                Ok(o) => o,
+                                Err(e) => {
+                                    eprintln!("Ignoring unusable pkg-config ({:?}): {:?}", path, e);
+                                    continue;
+                                }
+                            };
+                            if let Ok(out) = str::from_utf8(&out.stdout) {
+                                if out.contains("Pure-Perl") {
```

Comment:
```suggestion
                            if out.stdout.contains(b"Pure-Perl") {
```

maybe?

Reply:
`&[u8].contains` wants `&u8`, so that doesn't work